### PR TITLE
Spacing vars update

### PR DESCRIPTION
**Pull request by lyubomir-popov**
## Done

Review usage and fix some issues around spacing vars.

### CODE REVIEW

**Comment by anthonydillon** [comment]
Not sure if this is ready for review yet, but the few places of concern are breadcrumbs and notifications. 

Looks like lists have slightly less space below but that could be expected.

**Comment by lyubomir-popov** [comment]
@anthonydillon thanks for reviewing these. I fixed the notifications and the breadcrumbs.

**Comment by lyubomir-popov** [comment]
@anthonydillon for the lists, there's an invisible margin-bottom (which doesn't dstretch the example white background, but is there):
image

File: scss/_patterns_pull-quotes.scss [hunk 1]
```diff
@@ -65,7 +64,7 @@
   }
 
   .p-pull-quote {
-    padding: 0 $sph-outer * 2;
+    padding: 0 2rem;
```
Comment (sowasred2012) [inline scss/_patterns_pull-quotes.scss:67]: Is it intended that these values are hardcoded, rather than variables?

Comment (lyubomir-popov) [reply scss/_patterns_pull-quotes.scss:67]: These are offsets to fit a quote at the size it is used in each pattern, done by eyeballing what looks good basically - and this has nothing to do with keeping spacing consistent across components, as no other components use quotes at those font sizes. I just wanted that to be clear. 

For example, if you want to change that quote size, and respectively the spacing around it, you wouldn't expect anything outside the quote component to also change - therefore it shouldn't use $sph-outer.

And viceversa, if you changed `$sph-outer` so that components like the matrix gert a different spacing of the icon, you wouldn't expect the quote spacing to change as it is unrelated.

File: scss/_settings_grid.scss [hunk 2]
```diff
@@ -26,8 +26,9 @@ $grid-margin-widths: (
   large: $sp-unit * 3,
 ) !default;
 
-$grid-margin-width: $sph-outer--large !default;
-
+@include deprecate('3.0.0', 'Use the `$grid-margin-widths` map instead') {
```
Comment (bartaz) [inline scss/_settings_grid.scss:29]: Could you add a note about this to `update-guide-3.0.md` as well?

File: scss/_settings_colors.scss [hunk 3]
```diff
@@ -91,12 +91,9 @@ $colors--light-theme--border-low-contrast: rgba($color-x-dark, 0.1) !default;
 
 // Dark theme
 $colors--dark-theme--text-default: hsl(0, 0%, 100%) !default;
-$colors--dark-theme--nav-link-hover: $color-x-light !default;
-$colors--dark-theme--text-hover: hsl(0, 0%, 76%) !default; // minimum contrast on primary that satisfies contrast checker AA
 $colors--dark-theme--text-disabled: rgba($colors--dark-theme--text-default, 0.55) !default;
 $colors--dark-theme--text-muted: rgba($colors--dark-theme--text-default, $muted-text-opacity-amount) !default;
 $colors--dark-theme--text-inactive: rgba($colors--dark-theme--text-default, $inactive-text-opacity-amount) !default;
-$colors--dark-theme--nav-link-hover: $colors--dark-theme--text-hover !default;
```
Comment (bartaz) [inline scss/_settings_colors.scss:99]: This doesn't seem related to spacing. I guess it's just a drive-by clean up - are those not used anywhere? (I see that we don't have those in light theme).

File: scss/_patterns_pull-quotes.scss (hunk 1 above)
Comment (bartaz) [reply scss/_patterns_pull-quotes.scss:67]: If pull quote is a special case in terms of spacing we should comment on it in the code, so we know why vars are not used here.

Maybe worth adding some internal vars for this component only to be used instead?

File: scss/_settings_colors.scss (hunk 3 above)
Comment (lyubomir-popov) [reply scss/_settings_colors.scss:99]: nowhere in org:canonical-web-and-design

File: scss/_patterns_pull-quotes.scss (hunk 1 above)
Comment (lyubomir-popov) [reply scss/_patterns_pull-quotes.scss:67]: My goal initially was to decouple this from the vars as it is an incorrect coupling. Didn't use vars as it isn't used in more than one place. The correct way to do this is to measure the width of the qutaiton mark in ems, add a spacing var and use that. I'll add an issue for that.

Comment (lyubomir-popov) [reply scss/_patterns_pull-quotes.scss:67]: Issue created here https://github.com/canonical-web-and-design/vanilla-framework/issues/3821

File: scss/_settings_colors.scss (hunk 3 above)
Comment (lyubomir-popov) [reply scss/_settings_colors.scss:99]: @bartaz ready for another look.

File: scss/_patterns_breadcrumbs.scss [hunk 4]
```diff
@@ -5,6 +5,7 @@
   .p-breadcrumbs {
     list-style: none;
     margin: 0;
+    overflow-y: auto;
```
Comment (bartaz) [inline scss/_patterns_breadcrumbs.scss:8]: Why the change in overflow?

Comment (lyubomir-popov) [reply scss/_patterns_breadcrumbs.scss:8]: to make sure the margin-bottom of children is covered by the height of the parent. Otherwise it still applies, but it looks like it doesn't - the background of the example doesn't extend to cover it.

Comment (lyubomir-popov) [reply scss/_patterns_breadcrumbs.scss:8]: This should also be applied to the body, otherwise many components look like they have no margin-bottom, as @anthonydillon pointed out further up in this thread.

Comment (bartaz) [reply scss/_patterns_breadcrumbs.scss:8]: I don't think we should apply it for the sake of making examples look "better", we would need to apply this to every component in Vanilla and it can have unexpected consequences (scrollbars appearing).

If we want to make margins visible in examples we need to find different solution - maybe by applying some styles to body only in the examples? But that's something for another PR.

I'd remove this overflow from here as it doesn't seem to be needed for correct styling of the component.

**Review by bartaz — APPROVED** [review]
LGTM
